server/apps/distributed-app/docker-compose-with-pulsar.yml

```diff
@@ -0,0 +1,97 @@
+version: '3'
+
+services:
+
+  james:
+    depends_on:
+      cassandra:
+        condition: service_healthy
+      opensearch:
+        condition: service_started
+      tika:
+        condition: service_started
+      rabbitmq:
+        condition: service_started
+      s3:
+        condition: service_started
+      pulsar:
+        condition: service_started
+    image: apache/james:distributed-latest
+    container_name: james
+    hostname: james.local
+    command:
+      - --generate-keystore
+    networks:
+      - james
+    ports:
+      - "80:80"
+      - "25:25"
+      - "110:110"
+      - "143:143"
+      - "465:465"
+      - "587:587"
+      - "993:993"
+      - "8000:8000"
+    volumes:
+      - ./sample-configuration/pulsar.properties:/root/conf/pulsar.properties
+      - ./sample-configuration/keystore:/root/conf/keystore
+
+  opensearch:
+    image: opensearchproject/opensearch:2.1.0
+    environment:
+      - discovery.type=single-node
+      - DISABLE_INSTALL_DEMO_CONFIG=true
+      - DISABLE_SECURITY_PLUGIN=true
+    networks:
+      james:
+        aliases:
+          - elasticsearch
```

Comment:
```suggestion
          - opensearch
```

Reply:
I bet it will not work, because the "default" conf in james still use "elasticsearch" hostname

Reply:
Can u explain why? I see in docker-compose.yml, the value is also elasticsearch

Reply:
> I bet it will not work, because the "default" conf in James still use "elasticsearch" hostname

Default `opensearch.properties` for the distributed app:

```
opensearch.masterHost=opensearch
opensearch.port=9200
```

I did a Ctrl+Shift+F to seek for `elasticsearch` usage and see no misusage. I think it should be ok.

BTW @thanhbv200585 this alias change actually comes from master, not yours, let's skip it for now.